arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -55,6 +55,11 @@ extern K_THREAD_STACK_DEFINE(z_main_stack, CONFIG_MAIN_STACK_SIZE);
 uint32_t z_arm_mpu_stack_guard_and_fpu_adjust(struct k_thread *thread);
 #endif
 
+#if defined(CONFIG_CODE_DATA_RELOCATION)
+extern char __sram_text_start[];
```

Comment:
Are these symbols defined anywhere?

Reply:
They get generated at build time and are placed in `build/zephyr/include/generated/linker_relocate.ld`

Reply:
This assumes we always relocate to the SRAM region, which isn't necessarily true. It's possible to relocate to other regions.